rules/alerting.go

```diff
@@ -276,6 +299,7 @@ func (r *AlertingRule) Eval(ctx context.Context, ts time.Time, query QueryFunc,
 		}
 
 		vec = append(vec, r.sample(a, ts))
+		vec = append(vec, r.forStateSample(a, ts, float64(a.ActiveAt.UnixNano())))
```

Comment:
Curious if the floating point inaccuracies will mess something later.

Reply:
Floating point actually doesn't cause much harm (maybe few nano seconds?). We actually lose some small info when we are converting nano to milli [here](https://github.com/prometheus/prometheus/pull/4061/files#diff-d23c7264c420b842612a4fe05e018217R594). Hence at max we may end up losing few milliseconds (1-2), which should be fine? 

In my tests when I did multiple restarts, I saw a divergence of max 2 milliseconds from the actual time.